ipatests/test_integration/test_sssd.py

```diff
@@ -108,3 +114,57 @@ def test_auth_cache_enabled_when_configured(self, user):
             assert self.is_auth_cached(self.users[user])
             time.sleep(10)
             assert not self.is_auth_cached(self.users[user])
+
+    @contextmanager
+    def filter_user_setup(self, user, provider='ipa'):
+        tasks.kinit_admin(self.master)
+        master = self.master
+        sssd_conf_backup = tasks.FileBackup(master, paths.SSSD_CONF)
+        if user['name'] == 'fakeuser':
+            provider = 'nss'
+        filter = {'filter_users': '{0}'.format(user['name'])}
+        try:
+            tasks.modify_sssd_conf(
+                master, master.domain.name, filter, provider)
+            yield
+        finally:
+            sssd_conf_backup.restore()
+            tasks.clear_sssd_cache(self.master)
+
+    def is_user_filtered(self, user):
+        log_file = '{0}/sssd_nss.log'.format(paths.VAR_LOG_SSSD_DIR)
+        logsize = tasks.get_logsize(self.master, log_file)
+        try:
+            self.master.run_command(['getent', 'passwd', user['name']],
+                                    raiseonerr=False)
+            test_id = self.master.run_command(['id', user['name']],
+                                              raiseonerr=False)
+            assert test_id.returncode != 0
+        finally:
```

Comment:
if previous check have failed (i.e. `id` or `getent` returned success), we do not need to contonue the test, so try-except is not needed here